src/relayer/Relayer.ts

```
@@ -26,7 +28,53 @@ export class Relayer {
     readonly logger: winston.Logger,
     readonly clients: RelayerClients,
     readonly config: RelayerConfig
-  ) {}
+  ) {
+    this.logger.debug({
+      at: "Relayer",
+      message: "Instantiated with message relay rules",
+      messageRules: config.messageRules,
+    });
+  }
+
+  // Temporary feature to allow controlled use of message transfers.
+  permitMessageRelay(deposit: Deposit, rules: MessageRelayRule[]): boolean {
+    const l1Token = this.clients.hubPoolClient.getL1TokenInfoForL2Token(deposit.originToken, deposit.originChainId);
+    if (!isDefined(l1Token)) {
+      return false;
+    }
+
+    // The recipient might have changed via a speed-up event.
+    const recipient = isDepositSpedUp(deposit) ? deposit.updatedRecipient : deposit.recipient;
+    const rule = rules.find((rule) => {
+      return (
+        deposit.depositor === rule.depositor &&
+        rule.originChainIds.includes(deposit.originChainId) &&
+        rule.destinationChainIds.includes(deposit.destinationChainId) &&
+        rule.recipients.includes(recipient) &&
+        rule.tokenSymbols.includes(l1Token.symbol)
+      );
+    });
+
+    if (!isDefined(rule)) {
+      return false;
+    }
+
+    const idx = rule.tokenSymbols.indexOf(l1Token.symbol);
+    assert(idx !== -1, `Unexpected MessageRelayRule format (${rule.name})`); // invariant
```

Should we assert here? If I understand correctly, this would unwind the stack. Should we have retry logic to mitigate against transient issues?

---

It's an invariant, so if idx evaluates to -1 then something is wrong with the config parsing. I think it's preferable to just fail hard and fast in that case.